qiskit_ibm_runtime/session.py

```diff
@@ -179,9 +176,16 @@ def run(
 
         return job
 
-    def close(self) -> None:
-        """Close the session."""
+    def cancel(self) -> None:
+        """Cancel all pending jobs in a session."""
         self._active = False
+        if self._session_id:
+            self._service._api_client.cancel_session(self._session_id)
+
+    def close(self) -> None:
```

Comment:
This function should set `self._active=False` as well (maybe rename `_active` since it's now confusing). The `_active` flag is only used to check whether you can run a job in a session, and you cannot once `session.close()` is called.